packages/mongodb-rag-ingest/src/sources/mongodb-university/makeMarkdownPagesForLearningPaths.ts

```diff
@@ -0,0 +1,38 @@
+import fs from 'fs';
+import Path from "path";
+import {
+  TiCatalogItem,
+} from "./MongoDbUniversityDataApiClient";
+
+
+const SRC_ROOT = Path.resolve(__dirname, "..");
+
+/**
+  Helper function to create Markdown files
+  for MongoDB University learning paths and courses.
+ */
+export function makeMarkdownFilesForLearningPaths({
+  tiCatalogItems,
+}: {
+  tiCatalogItems: TiCatalogItem[];
+}): void {
+
+
+  const lp_dir = Path.resolve(SRC_ROOT, "../../../../mongodb-uni/learning-paths")
+  for(const item of tiCatalogItems) {
+    if(item.learning_format == 'Learning Path' || item.learning_format == 'Course') {
+      const { name, description, slug, nested_content } = item
+      const title = `# ${name}`
+      const markdownContent = [title, description, `\n`]
+      if (nested_content) {
+        for(const nested of nested_content) {
+          const { name, duration, description, slug } = nested
+          const title = `#### ${name}`
+          const link = `[View Details] (https://learn.mongodb.com/courses/${slug})`
+          markdownContent.push(title, duration, description, link, `\n`)
+        }
+      }
+      fs.writeFileSync(`${lp_dir}/${slug}.md`, markdownContent.join('\n'));
+    }
```

Comment:
rather than write to here, you can just write directly from the API to the `pages` collection

Reply:
but moreover, this should be a stateless function that takes in an object in the shape of an API response and returns the MD page. then we can plug that into the mdbu data source